source/reflect.h2

```diff
@@ -367,16 +366,18 @@ function_declaration: @copyable type =
 
     make_virtual                 : (inout this) -> bool = n*.make_function_virtual();
 
-    add_initializer: (move this, source: std::string_view)
-      pre(!has_initializer())
-      pre(parent_is_type())
+    add_initializer: (inout this, source: std::string_view)
     = {
-        prefix := n*.pretty_print_visualize(0);
-        prefix.back() = ' '; // Remove final ';'.
-        p := get_parent().as_type();
-        p.add_member(prefix + source);
-        _ = p;
-        mark_for_removal_from_enclosing_type();
+        require( !has_initializer(), 
+                 "cannot add an initializer to a function that already has one");
+        require( parent_is_type(),
+                 "cannot add an initializer to a function that isn't in a type scope");
+
+        stmt := parse_statement(source);
```

Comment:
Not using contracts also means that the `parse_statement` will be executed regardless of the preconditions.
Contracts seems like the better tool here.

It would still be nice to integrate the custom error handling of the reflection API with contracts.

Reply:
I have similar issues in my code.
I'm using it as I would a contract without continuation.

Reply:
I may have mentioned this somewhere else,
but I also expect the string argument to be a lazy parameter.
I can't remember what I wrote,
but I expected the condition to guard an invalid use in the interpolated string.

Reply:
I remember now.
It was something like
```Cpp2
  require(!v.empty() || v.size() == 2, "(v.front())$");
```
which the expectation that the string remains unevaluated when `v` is empty.

Reply:
Oh right, I should be at least using `if`...`error` instead of bunding them in `require`. Will fix...

Reply:
> It would still be nice to integrate the custom error handling of the reflection API with contracts.

Hold that thought... I'm trying it now and discovering some cool things seem to already work.

Spoiler: `pre<this>(expr, msg)` already invokes `(*this).expects(expr, msg);`. I discovered that when I tried to add support for `<this>` as a special-cased name only to discover it already worked. And there doesn't even have to be a `contract_group` object around as `this` behaves as its own custom contract group... +1 for duck typing and generic code.

Reply:
However, this is making me realize there's a reason not to turn all those `requires` checks into preconditions or assertions: Contracts shouldn't be used for control flow. Today, the way to turn off contract checks is to install a do-nothing handler, but if in the future there's a way to turn off contract checks, it would disable not only the checks but also the compiler diagnostics that would be emitted.